src/js/events/MesosSummaryActions.js

```diff
@@ -17,7 +17,10 @@ var _historyServiceOnline = true;
 function testHistoryServerResponse(response) {
   // If the response is an empty object, that means something is whack
   // Fall back to making requests to Mesos
-  if (response === {}) {
+  // TODO (DCOS-7764): This should be improved to validate against a schema
+  if (response === {} ||
+      Array.isArray(response.frameworks) ||
```

Comment:
Also, with this code, we flag the history service as offline if 
- response is an object AND
  - response.frameworks is an array OR
  - response.slaves is an array.

I think a valid response from the history server is
 - response is an object AND
    - response.arrays is an array AND
    - response.frameworks is an array OR
 - response is an array of such objects

If that's right we should adjust the test to reflect it.